src/Mod/Draft/draftobjects/array.py

```diff
@@ -0,0 +1,260 @@
+        
+        _tip = "Specifies if copies must be fused (slower)"
+        obj.addProperty("App::PropertyBool", "Fuse",
+                        "Parameters",QT_TRANSLATE_NOOP("App::Property", _tip))
+        obj.Fuse = False
+        if self.use_link:
+            obj.addProperty("App::PropertyInteger","Count","Draft",'')
+            obj.addProperty("App::PropertyBool","ExpandArray","Draft",
+                    QT_TRANSLATE_NOOP("App::Property","Show array element as children object"))
+            obj.ExpandArray = False
+
+        obj.ArrayType = ['ortho','polar','circular']
+        obj.NumberX = 1
+        obj.NumberY = 1
+        obj.NumberZ = 1
+        obj.NumberPolar = 1
+        obj.IntervalX = App.Vector(1,0,0)
+        obj.IntervalY = App.Vector(0,1,0)
+        obj.IntervalZ = App.Vector(0,0,1)
+        obj.Angle = 360
+        obj.Axis = App.Vector(0,0,1)
+        obj.RadialDistance = 1.0
+        obj.TangentialDistance = 1.0
+        obj.NumberCircles = 2
+        obj.Symmetry = 1
+
+        super(Array, self).attach(obj)
+
+    def linkSetup(self,obj):
+        super(Array, self).linkSetup(obj)
+        obj.configLinkProperty(ElementCount='Count')
+        obj.setPropertyStatus('Count','Hidden')
+
+    def onChanged(self,obj,prop):
+        super(Array, self).onChanged(obj, prop)
+        if prop == "AxisReference":
+            if obj.AxisReference:
+                obj.setEditorMode("Center", 1)
+                obj.setEditorMode("Axis", 1)
+            else:
+                obj.setEditorMode("Center", 0)
+                obj.setEditorMode("Axis", 0)
+
+    def execute(self,obj):
+        if obj.Base:
+            pl = obj.Placement
+            axis = obj.Axis
+            center = obj.Center
+            if hasattr(obj,"AxisReference") and obj.AxisReference:
+                if hasattr(obj.AxisReference,"Placement"):
+                    axis = obj.AxisReference.Placement.Rotation * App.Vector(0,0,1)
+                    center = obj.AxisReference.Placement.Base
+                else:
+                    raise TypeError("AxisReference has no Placement attribute. Please select a different AxisReference.")
+            if obj.ArrayType == "ortho":
+                pls = self.rectArray(obj.Base.Placement,obj.IntervalX,obj.IntervalY,
+                                    obj.IntervalZ,obj.NumberX,obj.NumberY,obj.NumberZ)
+            elif obj.ArrayType == "circular":
+                pls = self.circArray(obj.Base.Placement,obj.RadialDistance,obj.TangentialDistance,
+                                     axis,center,obj.NumberCircles,obj.Symmetry)
+            else:
+                av = obj.IntervalAxis if hasattr(obj,"IntervalAxis") else None
+                pls = self.polarArray(obj.Base.Placement,center,obj.Angle.Value,obj.NumberPolar,axis,av)
+
+            return super(Array, self).buildShape(obj, pl, pls)
+
+    def rectArray(self,pl,xvector,yvector,zvector,xnum,ynum,znum):
+        import Part
```

Comment:
I'm not sure why the Part modue is imported here, it doesn't seem to do anything.